scripts/feature_tests/upgrade/upgrade_vars.sh

```diff
@@ -11,18 +11,23 @@ function get_latest_capm3_release() {
 }
 
 # CAPM3 release version which we upgrade from.
-export CAPM3RELEASE="v0.5.0"
-CAPM3_REL_TO_VERSION="$(get_latest_capm3_release)" || true
-# CAPM3 release version which we upgrade to.
-export CAPM3_REL_TO_VERSION
+export CAPM3RELEASE="v0.5.2"
+
+if get_latest_capm3_release | grep -q 'Already'; then
```

Comment:
does it mean that we are already in the latest release of the provider?  If yes, why not skip verifying it and go directly `get_latest_capm3_release()`
basically,
```
CAPM3_REL_TO_VERSION="$(get_latest_capm3_release)" || true
export CAPM3_REL_TO_VERSION
```

Reply:
If I am not wrong, if the CAPM3RELEASE is already the latest version, `get_latest_capm3_release` will not return the latest version, but the word "Already", which is not the thing we can assign to `CAPM3_REL_TO_VERSION`.

Reply:
thanks, didn't know that. Why don't we do it for `CAPI_REL_TO_VERSION`  though?

Reply:
@fmuyassarov the reason for checking  `if get_latest_capm3_release | grep -q 'Already';` is, when we do not have yet release to upgrade, we set CAPM3_REL_TO_VERSION to CAPM3RELEASE which is no-op. Once we have 2 releases of CAPM3, we can set  `CAPM3RELEASE=1.0.0` and `CAPM3_REL_TO_VERSION=1.0.1` i.e. We do not need it for `CAPI_REL_TO_VERSION` because there is already 2 releases available, so `CAPIRELEASE=1.0.0` and `CAPI_REL_TO_VERSION` would be 1.0.1 based on  `clusterctl upgrade plan | grep cluster-api | awk 'NR == 1 {print $5}'`

Reply:
this is a workaround for now, got it